src/main/java/core/basesyntax/FigureSupplier.java

```
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_RANDOM_VALUE = 10;
+    public static final String DEFAULT_COLOR = "WHITE";
+    public static final double DEFAULT_RADIUS = 10;
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                return new Square(color, random.nextInt(MAX_RANDOM_VALUE) + 1);
+            case 1:
+                return new Rectangle(color, random.nextInt(MAX_RANDOM_VALUE) + 1,
+                        random.nextInt(MAX_RANDOM_VALUE) + 1);
+            case 2:
+                return new RightTriangle(color, random.nextInt(MAX_RANDOM_VALUE) + 1,
+                        random.nextInt(MAX_RANDOM_VALUE) + 1);
+            case 3:
+                return new Circle(color, random.nextInt(MAX_RANDOM_VALUE) + 1);
+            case 4:
+                return new IsoscelesTrapezoid(color, random.nextInt(MAX_RANDOM_VALUE) + 1,
+                        random.nextInt(MAX_RANDOM_VALUE) + 1, random.nextInt(MAX_RANDOM_VALUE) + 1);
+            default:
+                return null; // should not happen

```

yes, it shouldn't happen - better practice is to throw exception explicitly like IllegalStateException instead of returning null silently